MIDAS/src/hardware/GPSSensor.cpp

```diff
@@ -17,17 +17,22 @@ ErrorCode GPSSensor::init() {
     return ErrorCode::NoError;
 }
 
+// GPS Coordinates are ddmm.mmmm, so please convert with some code
 GPS GPSSensor::read() {
     teseo.update();
     GPGGA_Info_t gpgga_message = teseo.getGPGGAData();
     GPRMC_Info_t gprmc_message = teseo.getGPRMCData();
-    // Divide latitude and longitude by 100 because NMEA data is ddmmm.mmm or something like that
     // Also check if it's north or south
-    float lat = gpgga_message.xyz.lat / 100.f * ((gpgga_message.xyz.ns == 'N') ? 1 : -1);
-    float lon = gpgga_message.xyz.lon / 100.f * ((gpgga_message.xyz.ns == 'E') ? 1 : -1);
+    float64_t lat = gpgga_message.xyz.lat;
+    float64_t lon = gpgga_message.xyz.lon;
+
+    lat = static_cast<int>(lat)/100 + std::fmod(lat, 100.)/60.;
+    lon = static_cast<int>(lon)/100 + std::fmod(lon, 100.)/60.;
+    lat *= (gpgga_message.xyz.ns == 'N') ? 1. : -1.;
+    lon *= (gpgga_message.xyz.ew == 'E') ? 1. : -1.;
     float alt = gpgga_message.xyz.alt;
     float v = gprmc_message.speed;
     uint16_t sat_count = gpgga_message.sats;
 
-    return GPS{lat, lon, alt, v, sat_count};
+    return GPS{static_cast<float>(lat), static_cast<float>(lon), alt, v, sat_count};
```

Comment:
Still cannot be float at any point, floating point precision * earth circumference = hundred meter + errors